src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -1330,61 +1365,77 @@ bool StorageReplicatedMergeTree::executeLogEntry(LogEntry & entry)
         return true;
     }
 
-    if (entry.type == LogEntry::GET_PART ||
-        entry.type == LogEntry::MERGE_PARTS ||
-        entry.type == LogEntry::MUTATE_PART)
+    if (entry.type == LogEntry::ATTACH_PART)
+    {
+        if (MutableDataPartPtr part = attachPartHelperFoundValidPart(entry); part)
+        {
+            Transaction transaction(*this);
+
+            if (renameTempPartAndAdd(part, nullptr, &transaction))
+                checkPartChecksumsAndCommit(transaction, part);
+
```

Comment:
Maybe write something to `system.part_log`?

Reply:
I mean that `system.part_log` should contain `NewPart` event for every attached part, even if part was found locally (and was not fetched).